libr/core/cmd_api.c

```diff
@@ -363,6 +371,105 @@ R_API RCmdStatus r_cmd_call_parsed_args(RCmd *cmd, RCmdParsedArgs *args) {
 	return res;
 }
 
+static char *inner_get_help(RCmd *cmd, RCmdDesc *cd) {
+	RStrBuf *sb = r_strbuf_new (cd->help.summary);
+	r_strbuf_append (sb, "\n");
+
+	void **it_cd;
+	size_t max_len = 0;
+	r_cmd_desc_children_foreach (cd, it_cd) {
+		RCmdDesc *child = *(RCmdDesc **)it_cd;
+		size_t name_len = strlen (child->name);
+		size_t args_len = strlen (child->help.args_str);
+		if (name_len + args_len > max_len) {
+			max_len = name_len + args_len;
+		}
+	}
+	r_cmd_desc_children_foreach (cd, it_cd) {
+		RCmdDesc *child = *(RCmdDesc **)it_cd;
+		size_t str_len = strlen (child->name) + strlen (child->help.args_str);
+		size_t padding = str_len < max_len ? max_len - str_len : 0;
+		r_strbuf_appendf (sb, "| %s %s %*s# %s\n", child->name, child->help.args_str, padding, "", child->help.summary);
+	}
+	return r_strbuf_drain (sb);
+}
+
+static char *argv_get_help(RCmd *cmd, RCmdDesc *cd, RCmdParsedArgs *a, int detail) {
+	RStrBuf *sb = r_strbuf_new (NULL);
+	const char **e;
+
+	switch (detail) {
+	case 1:
+		r_strbuf_appendf (sb, "%s %s # %s\n", cd->name, cd->help.args_str, cd->help.summary);
+		return r_strbuf_drain (sb);
+	case 2:
+		r_strbuf_appendf (sb, "%s %s # %s\n\n", cd->name, cd->help.args_str, cd->help.summary);
+		r_strbuf_appendf (sb, "%s\n\n", cd->help.description);
+		r_strbuf_appendf (sb, "Examples:\n", cd->help.description);
+		for (e = cd->help.examples; *e; e += 2) {
+			r_return_val_if_fail (*(e + 1), NULL);
+			r_strbuf_appendf (sb, "| %s # %s\n", *e, *(e + 1));
+		}
+		return r_strbuf_drain (sb);
+	default:
+		r_strbuf_free (sb);
+		return NULL;
+	}
+}
+
+static char *oldinput_get_help(RCmd *cmd, RCmdDesc *cd, RCmdParsedArgs *a) {
+	const char *s = NULL;
+	r_cons_push ();
+	RCmdStatus status = r_cmd_call_parsed_args (cmd, a);
+	if (status == R_CMD_STATUS_OK) {
+		s = r_cons_get_buffer ();
+	}
+	char *res = strdup (s? s: "");
```

Comment:
`char *res = strdup ((status == R_CMD_STATUS_OK)? r_cons_get_buffer (): ""): `

Reply:
i did it this way because `s` could be NULL.